### PR TITLE
New createspace facet

### PR DESCRIPTION
**Pull request by giuseppecrj**
This PR moves the creation of spaces to a separate facet called CreateSpace which will allow backwards compatibility of old createSpace functions while being able to update it with bug fixes.

### CODE REVIEW

File: contracts/scripts/deployments/diamonds/DeploySpaceFactory.s.sol [hunk 1]
```diff
@@ -212,6 +216,11 @@ contract DeploySpaceFactory is DiamondHelper, Deployer {
         legacyRuleEntitlement // legacyRuleEntitlement
       )
     );
+    addFacet(
```
Comment (jterzis) [inline contracts/scripts/deployments/diamonds/DeploySpaceFactory.s.sol:219]: Well need to add this facet in https://github.com/river-build/river/blob/f7c2207602f41d7c5fb55055403722f4de3ccded/contracts/scripts/deployments/diamonds/DeploySpaceFactory.s.sol#L300 for alpha interaction ci script to upgrade this facet when it changes.

Comment (giuseppecrj) [reply contracts/scripts/deployments/diamonds/DeploySpaceFactory.s.sol:219]: yep, done in latest push

**Review by jterzis — APPROVED** [review]

File: contracts/src/factory/facets/create/CreateSpace.sol [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+// contracts
+import {Facet} from "contracts/src/diamond/facets/Facet.sol";
+import {ArchitectBase} from "contracts/src/factory/facets/architect/ArchitectBase.sol";
+import {PausableBase} from "contracts/src/diamond/facets/pausable/PausableBase.sol";
+import {ReentrancyGuard} from "contracts/src/diamond/facets/reentrancy/ReentrancyGuard.sol";
+
+contract CreateSpaceFacet is
+  ICreateSpace,
+  ArchitectBase,
+  PausableBase,
```
Comment (shuhuiluo) [inline contracts/src/factory/facets/create/CreateSpace.sol:18]: What bothers me is all facets on a diamond use the same `PausableBase` lock while we may only want to pause certain facets.

**Review by shuhuiluo — APPROVED** [review]
Utack.

File: contracts/test/base/registry/RewardsDistribution.t.sol [hunk 3]
```diff
@@ -1360,7 +1362,7 @@ contract RewardsDistributionTest is
         memory everyoneSpaceInfo = _createEveryoneSpaceInfo(spaceName);
       everyoneSpaceInfo.membership.settings.pricingModule = fixedPricingModule;
 
-      address everyoneSpace = Architect(spaceFactory).createSpace(
+      address everyoneSpace = CreateSpaceFacet(spaceFactory).createSpace(
```
Comment (shuhuiluo) [inline contracts/test/base/registry/RewardsDistribution.t.sol:1365]:
```suggestion
      address everyoneSpace = ICreateSpace(spaceFactory).createSpace(
```
nit

File: contracts/test/fork/ForkSpace.t.sol [hunk 4]
```diff
@@ -10,14 +10,16 @@ import {IArchitect, IArchitectBase} from "contracts/src/factory/facets/architect
...
+
 contract ForkOmegaSpace is TestUtils, SpaceHelper, IArchitectBase {
   address founder;
   address space;
 
   address spaceFactory = 0x9978c826d93883701522d2CA645d5436e5654252;
   address spaceOwner = 0x2824D1235d1CbcA6d61C00C3ceeCB9155cd33a42;
 
-  IArchitect spaceArchitect = IArchitect(spaceFactory);
+  CreateSpaceFacet spaceArchitect = CreateSpaceFacet(spaceFactory);
```
Comment (shuhuiluo) [inline contracts/test/fork/ForkSpace.t.sol:22]:
```suggestion
  ICreateSpace spaceArchitect = ICreateSpace(spaceFactory);
```
nit

File: contracts/test/fork/ForkSpaceOwner.t.sol [hunk 5]
```diff
@@ -90,7 +92,7 @@ contract ForkSpaceOwner is
       .settings
       .pricingModule = 0x7E49Fcec32E060a3D710d568B249c0ED69f01005;
 
-    IArchitect spaceArchitect = IArchitect(spaceFactory);
+    CreateSpaceFacet spaceArchitect = CreateSpaceFacet(spaceFactory);
```
Comment (shuhuiluo) [inline contracts/test/fork/ForkSpaceOwner.t.sol:95]:
```suggestion
    ICreateSpace spaceArchitect = ICreateSpace(spaceFactory);
```
nit

File: contracts/test/spaces/BaseSetup.sol [hunk 6]
```diff
@@ -167,8 +168,10 @@ contract BaseSetup is TestUtils, SpaceHelper {
     everyoneSpaceInfo.membership.settings.pricingModule = fixedPricingModule;
 
     vm.startPrank(founder);
-    space = Architect(spaceFactory).createSpace(spaceInfo);
-    everyoneSpace = Architect(spaceFactory).createSpace(everyoneSpaceInfo);
+    space = CreateSpaceFacet(spaceFactory).createSpace(spaceInfo);
+    everyoneSpace = CreateSpaceFacet(spaceFactory).createSpace(
+      everyoneSpaceInfo
+    );
```
Comment (shuhuiluo) [inline contracts/test/spaces/BaseSetup.sol:174]:
```suggestion
    space = ICreateSpace(spaceFactory).createSpace(spaceInfo);
    everyoneSpace = ICreateSpace(spaceFactory).createSpace(
      everyoneSpaceInfo
    );
```
nit

File: contracts/src/factory/facets/create/CreateSpace.sol (hunk 2 above)
Comment (giuseppecrj) [reply contracts/src/factory/facets/create/CreateSpace.sol:18]: Good point, let's discuss when we encounter this issue.

**Review by shuhuiluo — APPROVED** [review]

**Pull request merged**